### PR TITLE
Cursor updating and scrolling

### PR DESCRIPTION
**Pull request by benmuth**
Remote insertions/deletions now update the local cursor position so that it stays in the correct place, making it nicer for multiple people to edit at the same time.

Moving the cursor off the edge of the editor window now shifts the window to stay with the cursor. This is a minimal implementation of scrolling, equivalent to the behavior of the [kilo](github.com/antirez/kilo) editor.

Both features seem to work, and I wrote a few tests, but there could definitely be edge cases I haven't thought of. I'm planning to keep adding tests.

### CODE REVIEW

**Comment by burntcarrot** [comment]
Is it ready for review? @benmuth

**Comment by benmuth** [comment]
> Is it ready for review? @benmuth

Yes, please take a look!

File: client/editor/editor.go [hunk 1]
```diff
@@ -90,25 +118,42 @@ func (e *Editor) AddRune(r rune) {
...
+		cy -= e.GetRowOff()
+	}
+
 	termbox.SetCursor(cx-1, cy-1)
+	// find the starting and ending row of the termbox window.
+	yStart := e.GetRowOff()
+	yEnd := yStart + e.GetHeight() - 1 // -1 accounts for the status bar
+
+	// find the starting ending column of the termbox window.
+	xStart := e.GetColOff()
```
Comment (burntcarrot) [inline client/editor/editor.go:145]: There's something wrong with the current implementation: I tried adding a long piece of text (~90 characters; which is greater than the average width of a small terminal window (80 characters)); the scroll works nicely, however there's an issue with the final pieces of text getting lost. I don't think that the CRDT has lost content; as this PR involves changes to the UI only.

This is a weird, unusual issue that I was unable to debug, could you please take a look at it? It could be reproduced easily by adding a line with content exceeding the width of the terminal.

Comment (burntcarrot) [reply client/editor/editor.go:145]: For example, if I added `really loooo......ooong text`, the first scroll works as intended, but subsequent scrolls lose some content. (`really loooo......ooong`; `text` is not visible now)

Comment (benmuth) [reply client/editor/editor.go:145]: I'll take a look! Are you copying and pasting or typing it all out?

Comment (burntcarrot) [reply client/editor/editor.go:145]: Typing it all out.

Comment (benmuth) [reply client/editor/editor.go:145]: I'm having trouble reproducing the losing text issue (though I found a couple others).

I'm typing 50+ characters past the edge of the terminal window, all in one line, scrolling all the way back to the beginning of the line with the arrow key, then scrolling all the way back to the end of the line, but all the text is still there for me.

Comment (burntcarrot) [reply client/editor/editor.go:145]: download

Sorry for the delay, the GIF generation took some time. Here's what I did.

Comment (benmuth) [reply client/editor/editor.go:145]: Alright, I pushed a change and a couple more tests. Hopefully it fixes the strange cursor behavior when moving between lines.

Comment (burntcarrot) [reply client/editor/editor.go:145]: Looks good now!

**Review by burntcarrot — COMMENTED** [review]
Looks good to me! Just a small enhancement: can we use a flag for now for disable/enable scrolling?

As most editors have the option to toggle the word-wrap behavior, it would be nice to have it here too. Let's use flags for now, I'm thinking of using a config file in the future.

**Review by burntcarrot — APPROVED** [review]
Awesome! Thanks for the PR! Merging.
